OpenRA.Game/Map/MapPreview.cs

```diff
@@ -273,16 +273,17 @@ public MapPreview(Map map, ModData modData)
 
 			innerData = new InnerData
 			{
-				MapFormat = map.MapFormat,
+				// TO DO: Check if this MapFormat retrieval is correct and working
+				MapFormat = Map.GetMapFormat(Package),
```

Comment:
GetMapFormat is only used to get the format before yaml parsing. You are using it after

Reply:
Fixed, this now uses a cast of `IMap` as with the other attributes.

Reply:
Still unresolved. Keep the original implementation